src/build_workflow/build_target.py

```diff
@@ -27,7 +27,7 @@ def __init__(
         build_id=None,
         output_dir="artifacts",
     ):
-        self.build_id = os.getenv("OPENSEARCH_BUILD_ID") or build_id or uuid.uuid4().hex
+        self.build_id = os.getenv("OPENSEARCH_BUILD_ID") or os.getenv("OPENSEARCH_DASHBOARDS_BUILD_ID") or build_id or uuid.uuid4().hex
```

Comment:
This definitely tells me that we should merge these into a single variable. Then I think there was a comment on why we rename `BUILD_NUMBER` to `OPENSEARCH_DASHBOARDS_BUILD_ID`. Feels like all these things could be collapsed into a single `BUILD_NUMBER`. Wouldn't it make things a lot simpler?

Plus we'll also refactor Jenkinsfile into a shared library at some point I'm sure to avoid all this code duplication.